lib/index.ts

```diff
@@ -326,3 +328,114 @@ export function main() {
     fs.writeFileSync(outPath, generatedCode);
   }
 }
+
+const macroHandler: MacroHandler = params => {
```

Comment:
This whole function is outside my competence. I can review enough to approve it, but if you'd like any meaningful input, perhaps can ask someone familiar with babel plugins.

Reply:
Unfortunately I don't know anyone personally who can help. Perhaps we can do a call-out to users somehow? A pinned issue?

I'm not really worried that there will be any major problems with this implementation, but really two heads are better than one, and there might be some way to really improve (or fix) this implementation and make it better.

Reply:
Users will report any issues I guess.